### PR TITLE
Exosuit: Ripley

### PR DESCRIPTION
**Pull request by EmoGarbage404**
## About the PR 
Adds ripley as a mech able to be created through science. Also adds the hydraulic clamp for picking up and moving stuff.

**Screenshots**

https://user-images.githubusercontent.com/98561806/205476477-2322f5c9-334b-4a4a-806f-0e31d3796d46.mp4

image
image

image
image

image

**Changelog**

:cl:
- add: Added Ripley, Cargo's favorite mechanized machine of destruc-- I mean, mechanized machine of honest cargo transfer.

### CODE REVIEW

**Comment by EmoGarbage404** [comment]
it's a little known fact that i open these prs solely for flexing big features before i actually code them

**Comment by LarryRussian** [comment]
wow
We are waiting for all the Russian community.

**Comment by EmoGarbage404** [comment]
oh yeah, if anyone wants to help me out with construction graphs and other such things, I would appreciate it.

**Comment by Easypoller** [comment]
I'll help with stuff :)

**Comment by EmoGarbage404** [comment]
someone find me those tg sprites for cycling equipment and opening the ui i know they exist i'm going insane.

**Comment by mirrorcult** [comment]
[actions_mecha.zip](https://github.com/space-wizards/space-station-14/files/10079935/actions_mecha.zip)

**Comment by Easypoller** [comment]
I have been fired :C

**Comment by EmoGarbage404** [comment]
Ready for review

**Comment by PixelTheKermit** [comment]
this will be fun

File: Content.Client/Mech/MechAssemblyVisualsComponent.cs [hunk 1]
```diff
@@ -0,0 +1,11 @@
+﻿namespace Content.Client.Mech;
+
+/// <summary>
+/// This is used for...
+/// </summary>
```
Comment (moonheart08) [inline Content.Client/Mech/MechAssemblyVisualsComponent.cs:5]: .

File: Content.Server/Mech/Equipment/EntitySystems/MechGrabberSystem.cs [hunk 2]
```diff
@@ -0,0 +1,173 @@
...
+        var mech = equipmentComponent.EquipmentOwner.Value;
+
+        var targetCoords = new EntityCoordinates(mech, component.DepositOffset);
+        if (!_interaction.InRangeUnobstructed(mech, targetCoords))
+            return;
+
+        if (!component.ItemContainer.Contains(msg.Item))
+            return;
+
+        Logger.Debug("ok now remove it");
```
Comment (moonheart08) [inline Content.Server/Mech/Equipment/EntitySystems/MechGrabberSystem.cs:55]: ok now remove it

File: Content.Server/Mech/Equipment/EntitySystems/MechGrabberSystem.cs [hunk 3]
```diff
@@ -0,0 +1,173 @@
...
+        if (xform.Anchored || HasComp<WallMountComponent>(args.Target.Value))
+            return;
+
+        if (component.ItemContainer.ContainedEntities.Count >= component.MaxContents)
+            return;
+
+        if (!TryComp<MechComponent>(args.User, out var mech))
+            return;
+
+        if (mech.Energy + component.EnergyPerGrab < 0)
```
Comment (moonheart08) [inline Content.Server/Mech/Equipment/EntitySystems/MechGrabberSystem.cs:122]: Is this meant to be a subtraction?

**Review by moonheart08 — CHANGES_REQUESTED** [review]
Please document every system, public function, and public field/property. Most only need brief documentation but never-the-less.

File: Content.Server/Mech/Equipment/EntitySystems/MechGrabberSystem.cs (hunk 3 above)
Comment (EmoGarbage404) [reply Content.Server/Mech/Equipment/EntitySystems/MechGrabberSystem.cs:122]: no. The energy per grab is a negative number. I'll rename it to be a bit clearer

**Review by moonheart08 — APPROVED** [review]

**Pull request merged**